openquake/calculators/event_based_risk.py

```diff
@@ -59,14 +59,21 @@ def build_agg_curve(r_data, insured_losses, ses_ratio, curve_resolution,
     for r, data in r_data:
         if len(data) == 0:  # realization with no losses
             continue
-        E, L, I = data['loss'].shape
+        E, L = data['loss'].shape[:2]
         for l in range(L):
-            for i in range(insured_losses + 1):
-                # if insured_losses is False, don't build them
+            losses, poes = scientific.event_based(
+                data['loss'][:, l, 0], ses_ratio, curve_resolution)
+            avg = scientific.average_loss((losses, poes))
+            result[l, r, 'losses'] = losses
+            result[l, r, 'poes'] = poes
+            result[l, r, 'avg'] = avg
+            if insured_losses:
                 losses, poes = scientific.event_based(
```

Comment:
please rename `losses`, `poes` and `avg` into `losses_ins`, `poes_ins` and `avg_ins`